app/src/pages/market_sections/market_home_container.tsx

```diff
@@ -140,26 +147,30 @@ const MarketHomeContainer: React.FC = () => {
   let typeRoute = location.pathname.split('/type/')[1]
   if (typeRoute) typeRoute = typeRoute.split('/')[0]
 
-  const stateFilter = location.search.includes('state')
-  let stateRoute = location.search.split('state=')[1]
-  if (stateRoute) stateRoute = stateRoute.split('&')[0]
-
   const searchFilter = location.search.includes('tag')
   let searchRoute = location.search.split('tag=')[1]
   if (searchRoute) searchRoute = searchRoute.split('&')[0]
 
   let sortParam: Maybe<MarketsSortCriteria> = stateRoute === 'MY_MARKETS' ? 'openingTimestamp' : 'usdLiquidityParameter'
-  if (sortRoute === '24h-volume') {
-    sortParam = `sort24HourVolume${Math.floor(Date.now() / (1000 * 60 * 60)) % 24}` as MarketsSortCriteria
-  } else if (sortRoute === 'volume') {
-    sortParam = 'usdVolume'
-  } else if (sortRoute === 'newest') {
-    sortParam = 'creationTimestamp'
-  } else if (sortRoute === 'ending') {
-    sortParam = 'openingTimestamp'
-    sortDirection = 'asc'
-  } else if (sortRoute === 'liquidity') {
-    sortParam = 'usdLiquidityParameter'
+  let sortIndex: number = stateRoute === 'MY_MARKETS' ? 1 : 2
+  if (stateRoute !== 'MY_MARKETS') {
+    if (sortRoute === '24h-volume') {
+      sortIndex = 0
+      sortParam = `sort24HourVolume${Math.floor(Date.now() / (1000 * 60 * 60)) % 24}` as MarketsSortCriteria
+    } else if (sortRoute === 'volume') {
+      sortIndex = 1
+      sortParam = 'usdVolume'
+    } else if (sortRoute === 'liquidity') {
+      sortIndex = 2
+      sortParam = 'usdLiquidityParameter'
+    } else if (sortRoute === 'newest') {
+      sortIndex = 3
+      sortParam = 'creationTimestamp'
+    } else if (sortRoute === 'ending') {
+      sortIndex = 4
+      sortParam = 'openingTimestamp'
+      sortDirection = 'asc'
+    }
```

Comment:
@KadenZipfel Wondering is it maybe more efficient to make this into switch/case statement?